modules/swagger-models/src/main/java/io/swagger/v3/oas/models/media/MediaType.java

```diff
@@ -93,10 +95,11 @@ public Object getExample() {
 
     public void setExample(Object example) {
         this.example = example;
+        exampleSetFlag = true;
```

Comment:
see how this is done in [Schema](https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-models/src/main/java/io/swagger/v3/oas/models/media/Schema.java#L690-L700) to correctly handle all scenarios